src/main/scala/com/databricks/spark/csv/package.scala

```diff
@@ -43,21 +43,23 @@ package object csv {
   }
 
   implicit class CsvSchemaRDD(dataFrame: DataFrame) {
-    def saveAsCsvFile(path: String): Unit = {
+    def saveAsCsvFile(path: String, parameters: Map[String, String] = Map()): Unit = {
       // TODO(hossein): For nested types, we may want to perform special work
-      val header = dataFrame.columns.map(c => s""""$c"""").mkString(",")
+      val delimiter = parameters.getOrElse("delimiter", ",")
+      val generateHeader = parameters.getOrElse("generateHeader", "false").toBoolean
+      val header = dataFrame.columns.map(c => s""""$c"""").mkString(delimiter)
```

Comment:
We can but it will be a bit hacky. We can't move this line inside mapPartitions (it uses `dataFrame`), but I can use an if here, and leave it empty if `header` is false. I guess this is tolerable, I'll make the change.